CONTRIBUTING.md

```diff
@@ -26,133 +26,169 @@ We specifically do not merge PRs **without prior issues** that:
+> [!TIP]
+> Replace the `GOOS` and `GOARCH` values with your target platform if you wish to cross-compile. 
 
-**⚠️ Important:** Please avoid working on features or refactor without [an `accepted` issue](https://github.com/opentofu/opentofu/issues?q=is%3Aopen+is%3Aissue+label%3Aaccepted). OpenTofu is a large and complex project and every change needs careful consideration. We cannot merge non-bug pull requests without first having a discussion about them, no matter how trivial the issue may seem.
+---
 
-In order for a code change to be accepted, you'll also have to accept the Developer Certificate of Origin (DCO). It's very lightweight, and you can find it [here](https://developercertificate.org). Accepting is accomplished by signing off on your commits, you can do this by adding a `Signed-off-by` line to your commit message, like here:
-```
-This is my commit message
+### Running tests
+
+Similar to builds, you can use the `go test` command to run tests. To run the entire test suite, please run the following command in your OpenTofu source directory:
 
-Signed-off-by: Random Developer <random@developer.example.org>
+```sh
+go test ./...
 ```
-Git has a built-in flag to append this line automatically:
+
+Alternatively, you can also run the `go test` command in the package you are currently working on.
+
+---
+
+### Updating the changelog
+
+We are keeping track of the changes to OpenTofu in the [CHANGELOG.md](CHANGELOG.md) file. Please update it when you add features or fix bugs in OpeTofu.
+
+---
+
+### Signing off your commits
+
+When you contribute code to OpenTofu, we require you to add a [Developer Certificate of Origin](https://developercertificate.org/) sign-off. Please read the DCO carefully before you proceed and only contribute code you have written yourself. Please do not add code that you have not written from scratch yourself without discussing it in the related issue first.
+
+The simplest way to add a sign-off is to use the `-s` command when you commit:
+
 ```
-~> git commit -s -m 'This is my commit message'
+git commit -s -m "My commit message"
 ```
 
-You can find more details about the DCO checker in the [DCO app repo](https://github.com/dcoapp/app).
+> [!IMPORTANT]
+> Make sure your `user.name` and `user.email` setting in Git matches your GitHub settings. This will allow the automated DCO check to pass and avoid delays when merging your PR.
 
-Additionally, please update [the changelog](CHANGELOG.md) if you're making any user-facing changes.
+> [!TIP]
+> Have you forgotten your sign-off? You can use `git rebase --signoff` to add it after the fact.
 
```

Comment:
It may be worth mentioning that this works as a check on a pull request, and if it fails they can check the details of the failing check for more information about why.